core/raw/metadata.cpp

```diff
@@ -123,16 +123,22 @@ const QStringList& asciiTags()
     return asciiNames;
 }
 
-const QStringList& niceTags()
+const QStringList& niceTags(bool restricted)
 {
     QStringList ret;
-    for (int i=0; i<metaDefs.size(); i++)
-        ret.append(niceTag(i));
+    for (int i=0; i<metaDefs.size(); i++) {
+        if (restricted) {
+            if (getMetaMode(i) !=  metaMode::CONSTANT)
+                ret.append(niceTag(i));
+        }
+        else {
+            ret.append(niceTag(i));
+        }
+    }
```

Comment:
could be done more simply, using the combined logical expression

    if ( !(restricted && getMetaMode(i) ==  metaMode::CONSTANT) )